src/main/java/com/wire/bots/sdk/server/resources/MessageResourceBase.java

```diff
@@ -108,7 +109,11 @@ protected void handleMessage(Payload payload, WireClient client) throws Exceptio
             case "user.connection": {
                 Logger.debug("user.connection: bot: %s, status: %s", botId, payload.connection.status);
                 if (payload.connection.status.equals("pending")) {
-                    client.acceptConnection(payload.connection.to);
+                    UUID convId = client.acceptConnection(payload.connection.to);
+
+                    try (WireClient c = ((UserClientRepo) repo).getWireClient(UUID.fromString(botId), convId)) {
```

Comment:
This is probably not needed as `conversation.create` event is created and sent by the Backend. 
You can turn on DEBUG logging mode and monitor what events are sent upon `connection.accept`
```
logging:
  level: INFO
  loggers:
    "com.wire.bots.sdk.tools.logger": DEBUG
```

Reply:
During my tests no `conversation.create` event was received after the pending connection request was received. Otherwise the existing logic that handles `conversation.create` events would have worked for me without any modifications.

Reply:
You are right. Just ran the test. Nothing is sent to Bot, indeed. Event is sent to the other party. What you could do is to treat new `onConnectRequest` listener as `onConversationCreated`
If you need to, let's say, 'Say There!' to new User:
```
 @Override
    public void onConnectionRequest(WireClient client, String status, String to) {
        super.onConnectionRequest(client, status, to);
        client.sendText("Hi There!");
    }
```

You see, in this case the conversation was create BEFORE the connect request was accept.
We are just accepting the req and sending first msg. No need for creating new WireClient object.
I just need to check if the ConversationID is correctly set

Reply:
Since the payload looks like:
```
{
    "id": "c84f3e9d-367e-11e9-8002-22000a798f87", 
    "payload": [
        {
            "connection": {
                "conversation": "afd49687-f79f-45f8-add1-d72a5822a3f2", 
                "from": "3d49ca62-17b0-4eaa-a648-04973d1d1a9b", 
                "last_update": "2019-02-22T08:49:36.576Z", 
                "message": " ", 
                "status": "pending", 
                "to": "728acc25-dfef-474e-8789-d2931b058957"
            }, 
            "type": "user.connection", 
            "user": {
                "name": "Dejan"
            }
        }
    ], 
    "transient": false
}
```

We need to add `conversation` UUID to `Connection` object:
```
@JsonIgnoreProperties(ignoreUnknown = true)
    public static class Connection {
        @JsonProperty
        public String status;

        @JsonProperty
        public String to;

        @JsonProperty
        public UUID conversation;
    }
```
and then this UUID needs to be handled slightly differently in: `Endpoint.onMessage` like:
```
...
case "user.connection":
    userMessageResource.onNewMessage(user.getUserId(), payload.connection.conversation, payload);
    break;
...
```

Reply:
I completely missed that the conversationId is already included in the pending connection request. Thanks for the clarification! I've updated the code according to your suggestions in d982fd1e1d300796895af1259c875c879cdb5a65 and removed the now unnecessary response parsing that I added before.

Reply:
Shall we call `handler.onNewConversation(client);` just after `handler.onConnectReques(client, ...);` in `handleMessage`? In that case Dev does not have to override the `onNewConversation` and can keep the Hello logic in `onNewConversation`

Reply:
This would be an option if there are no other situations where a connection request event is sent by the server. I would assume that such an event is also received when the bot user initiates the connection request and the other user accepts or ignores the request? So if we add the additional `handler.onNewConversation(client)` call, it should only be called when the status is `pending`?

Reply:
Correct. `onConnectReques` should return `true` only if `status` was `pending` and the connect req was accepted. Then the code would look like:
```
if(handler.onConnectReques(client, ...))
   handler.onNewConversation(client);
```
We don't have to worry about bots sending connect reqs as bots cannot do that ;)
If you bot that runs in `UserMode` wants to send a connect request then the `status` would be `sent` and the other party would get a `pending`